Arrowgene.Ddon.GameServer/Characters/WalletManager.cs

```diff
@@ -47,15 +38,19 @@ public bool AddToWalletNtc(Client Client, Character Character, WalletType Type,
             return true;
         }
 
-        public bool AddToWallet(Character Character, WalletType Type, uint Amount)
+        public CDataUpdateWalletPoint AddToWallet(Character Character, WalletType Type, int Amount)
         {
             CDataWalletPoint Wallet = Character.WalletPointList.Where(wp => wp.Type == Type).Single();
 
-            Wallet.Value += Amount;
+            Wallet.Value = (uint) Math.Max(0, (int)Wallet.Value + Amount);
```

Comment:
Yes, it's to allow substracting while preventing underflows. It shouldnt ever happen but in case it happens its a decent last resort guard. IF youd rather have it raise an exception it can be done too.
It doesnt return the packet but rather the structure that goes into the packet. That way you can add more than one operation into the same packet

Reply:
We have a subtract from wallet function. Also, in this implementation, if someone has max money and then they add more, they will be left with 0 money instead of max money.

Reply:
Addressed